### PR TITLE
Make log strings more readable as structured ones.

### PR DESCRIPTION
**Pull request by joe94**

### CODE REVIEW

File: src/caduceus/http.go [hunk 1]
```diff
@@ -35,16 +37,16 @@ func (sh *ServerHandler) ServeHTTP(response http.ResponseWriter, request *http.R
 		response.WriteHeader(http.StatusBadRequest)
 		response.Write([]byte(fmt.Sprintf("Unsupported method \"%s\"... Caduceus only supports \"POST\" method.\n", request.Method)))
 
-		debugLog.Log(messageKey, "Unsupported method", "method", request.Method, "explanation",
-		"Caduceus only supports \"POST\" method.")
+		errorLog.Log(errorKey, "Unsupported method", "method", request.Method, messageKey,
```
Comment (joe94) [inline src/caduceus/http.go]: @johnabass This log did not seem a debug level (as it was before switching to go-kit) one but rather that of an error. Correct if I am wrong.

File: src/caduceus/http.go [hunk 2]
```diff
@@ -35,16 +37,16 @@ func (sh *ServerHandler) ServeHTTP(response http.ResponseWriter, request *http.R
...
+		"Caduceus only supports POST method.")
 		return
 	}
 
 	payload, err := ioutil.ReadAll(request.Body)
 	if err != nil {
 		response.WriteHeader(http.StatusBadRequest)
 		response.Write([]byte(fmt.Sprintf("Unable to retrieve the request body: %s.\n", err.Error)))
-		debugLog.Log(messageKey, "Unable to retrieve the request body.", logging.ErrorKey(), err.Error)
+		errorLog.Log(messageKey, "Unable to retrieve the request body.", errorKey, err.Error)
```
Comment (joe94) [inline src/caduceus/http.go]: @johnabass same as above.

File: src/caduceus/http.go [hunk 3]
```diff
@@ -62,8 +64,8 @@ func (sh *ServerHandler) ServeHTTP(response http.ResponseWriter, request *http.R
 				response.Write([]byte(fmt.Sprintf("Only Content-Type values of \"application/json\" or " +
 					"\"application/msgpack\" are supported got: [%s].\n", value)))
 
-				debugLog.Log("explanation", "explanation","Only Content-Type values of \"application/json\" or " +
-					"\"application/msgpack\" are supported.", "actual", value)
+				errorLog.Log(messageKey,"Only Content-Type values 'application/json' or 'application/msgpack' are " +
```
Comment (joe94) [inline src/caduceus/http.go]: Same ^

File: src/caduceus/outboundSender.go [hunk 4]
```diff
@@ -503,7 +500,7 @@ func (obs *CaduceusOutboundSender) queueOverflow() {
 						"notification", obs.listener.FailureURL)
 				} else {
 					// Success
-					errorLog.Log("Able to send cut-off notification", "url", obs.listener.FailureURL,
+					logging.Info(obs.logger).Log("Able to send cut-off notification", "url", obs.listener.FailureURL,
```
Comment (joe94) [inline src/caduceus/outboundSender.go]: @johnabass It seemed odd that a success case was being logged at error level.

File: src/caduceus/http.go (hunk 1 above)
Comment (johnabass) [reply src/caduceus/http.go]: This log should be removed entirely.  The mapping for the handler should prevent unsupported methods up front, so code should never get into this case.

File: src/caduceus/http.go (hunk 2 above)
Comment (johnabass) [reply src/caduceus/http.go]: This log works fine, but code should not attempt to write anything to the response.  If a request body occurs, that indicates a low-level socket I/O error and the response/request pair should be abandoned.

File: src/caduceus/http.go (hunk 3 above)
Comment (johnabass) [reply src/caduceus/http.go]: Again, the mapping should prevent content types that are unsupported from reaching this block of code.  Remove the log and verify that there's a `Headers("Content-Type", ...)` mapper.

File: src/caduceus/http.go [hunk 5]
```diff
@@ -118,6 +120,7 @@ type ProfileHandler struct {
 // that the main handler has successfully dealt with
 func (ph *ProfileHandler) ServeHTTP(response http.ResponseWriter, request *http.Request) {
 	logging.Info(ph.Logger).Log(logging.MessageKey(), "Receiving request for server stats...")
```
Comment (johnabass) [inline src/caduceus/http.go]: This is way too much logging.  Remove this entirely.  If we want request logs, that should be handled in middleware.

File: src/caduceus/senderWrapper_test.go [hunk 6]
```diff
@@ -46,7 +47,7 @@ func (t *swTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 }
 
 func swGetLogger() log.Logger {
```
Comment (johnabass) [inline src/caduceus/senderWrapper_test.go]: This function is a meaningless abstraction.  Just replace calls to `swGetLogger` to `logging.DefaultLogger()`

File: src/caduceus/http.go (hunk 2 above)
Comment (joe94) [reply src/caduceus/http.go]: Understood. Removed lines 47-48.

File: src/caduceus/http.go (hunk 1 above)
Comment (joe94) [reply src/caduceus/http.go]: Addressed.

File: src/caduceus/http.go (hunk 3 above)
Comment (joe94) [reply src/caduceus/http.go]: Addressed.

**Pull request merged**